File: connecttoapilibrary/src/main/java/com/sec/connecttoapilibrary/qtapitest/liveRadioData/StationProgram.java
package com.sec.connecttoapilibrary.qtapitest.liveRadioData;

import java.io.Serializable;

/**
 * Created by SRC-TJ-MM-BinYang on 2018/1/16.
 */

public class StationProgram implements Serializable {
    private int id ;//节目id
    private String start_time ;//播放开始时间
    private String end_time ;//播放结束时间
    private int duration ;//播放时长，单位秒
    private String title ;//标题
    private String type ;//节目类型
    private int channel_id ;//所属电台id
    private int program_id ;//节目唯一标示符
    private Broadcasters[] broadcasters;

    private int broadcasters_num = 0;

    public void newBroadcasters(int num){
        broadcasters_num = num;
        broadcasters = new Broadcasters[num];
    }

    public Broadcasters[] getBroadcaster(){
        return broadcasters;
    }

    public int getBroadcastersNum(){
        return broadcasters_num;
    }

    public void setProgramId(int id){
        this.id = id;
    }
    public int getProgramId(){
        return id;
    }

    public void setProgramStart_time(String start_time){
        this.start_time = start_time;
    }
    public String getProgramStart_time(){
        return start_time;
    }

    public void setProgramEnd_Time(String end_time){
        this.end_time = end_time;
    }
    public String getProgramEnd_Time(){
        return end_time;
    }

    public void setProgramDuration(int duration){
        this.duration = duration;
    }
    public int getProgramDuration(){
        return duration;
    }

    public void setProgramTitle(String title){
        this.title = title;
    }
    public String getProgramTitle(){
        return title;
    }

    public void setProgramType(String type){
        this.type = type;
    }
    public String getProgramType(){
        return type;
    }

    public void setProgramChannel_Id(int channel_id){
        this.channel_id = channel_id;
    }
    public int getProgramChannel_Id(){
        return channel_id;
    }

    public void setProgramProgram_Id(int program_id){
        this.program_id = program_id;
    }
    public int getProgramProgram_Id(){
        return program_id;
    }

    public static class Broadcasters {
        /*also can get like wei bo or QQ id if has the info*/
        public int id;
        public String username;
        public void setBroadcastersName(String username){
            this.username = username;
        }
        public String getBroadcastersName(){
            return username;
        }
    }
}

File: myonlinefm/src/main/java/com/sec/myonlinefm/data/StationProgram.kt
package com.sec.myonlinefm.data

/**
 * Created by SRC-TJ-MM-BinYang on 2018/1/16.
 */
class StationProgram {
    private var id //节目id
            = 0
    private var start_time //播放开始时间
            : String? = null
    private var end_time //播放结束时间
            : String? = null
    private var duration //播放时长，单位秒
            = 0
    private var title //标题
            : String? = null
    private var type //节目类型
            : String? = null
    private var channel_id //所属电台id
            = 0
    private var program_id //节目唯一标示符
            = 0
    private var broadcasters: Array<Broadcasters?>? = null
    private var broadcasters_num = 0
    fun newBroadcasters(num: Int) {
        broadcasters_num = num
        broadcasters = arrayOfNulls<Broadcasters?>(num)
    }

    fun getBroadcaster(): Array<Broadcasters?>? {
        return broadcasters
    }

    fun getBroadcastersNum(): Int {
        return broadcasters_num
    }

    fun setProgramId(id: Int) {
        this.id = id
    }

    fun getProgramId(): Int {
        return id
    }

    fun setProgramStart_time(start_time: String?) {
        this.start_time = start_time
    }

    fun getProgramStart_time(): String? {
        return start_time
    }

    fun setProgramEnd_Time(end_time: String?) {
        this.end_time = end_time
    }

    fun getProgramEnd_Time(): String? {
        return end_time
    }

    fun setProgramDuration(duration: Int) {
        this.duration = duration
    }

    fun getProgramDuration(): Int {
        return duration
    }

    fun setProgramTitle(title: String?) {
        this.title = title
    }

    fun getProgramTitle(): String? {
        return title
    }

    fun setProgramType(type: String?) {
        this.type = type
    }

    fun getProgramType(): String? {
        return type
    }

    fun setProgramChannel_Id(channel_id: Int) {
        this.channel_id = channel_id
    }

    fun getProgramChannel_Id(): Int {
        return channel_id
    }

    fun setProgramSelf_Id(program_id: Int) {
        this.program_id = program_id
    }

    fun getProgramSelf_Id(): Int {
        return program_id
    }

    class Broadcasters {
        /*also can get like wei bo or QQ id if has the info*/
        var id = 0
        var username: String? = null
        fun setBroadcastersName(username: String?) {
            this.username = username
        }

        fun getBroadcastersName(): String? {
            return username
        }
    }
}